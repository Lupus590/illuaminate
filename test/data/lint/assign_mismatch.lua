-- config: (lint (only var:unbalanced-assign))

local x, y = 0 -- ok
local x, y = 0, 0, 0

x, y = 0
x, y = 0, 0, 0
x, y = ...
