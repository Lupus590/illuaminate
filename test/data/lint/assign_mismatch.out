assign_mismatch.lua:[4:0-4:20]: Right-hand side of assignment has more values than left hand side expects. [var:unbalanced-assign]
   │
 4 │ local x, y = 0, 0, 0
   │ ^^^^^^^^^^^^^^^^^^^^
assign_mismatch.lua:[6:0-6:8]: Right-hand side of assignment has less values than left hand side expects. [var:unbalanced-assign]
   │
 6 │ x, y = 0
   │ ^^^^^^^^
assign_mismatch.lua:[7:0-7:8]: Right-hand side of assignment has more values than left hand side expects. [var:unbalanced-assign]
   │
 7 │ x, y = 0, 0, 0
   │ ^^^^^^^^
Compilation succeeded with 3 warnings
